### PR TITLE
Quality fixes

### PR DESCRIPTION
**Pull request by dylburger**
### <samp>🤖 Generated by Copilot at 9535c5e</samp>

This pull request updates the code generation module and templates to support different types of components, improve the code style and readability, and fix some minor issues. It adds a `component_type` argument to the `generate_code` function and uses the `templates` module to select the appropriate checks and templates based on the component type. It also applies PEP 8 formatting and adds headers and comments to the code and templates. It updates the nodejs version in the `.tool-versions` file and the logging level in the `config.py` file. It removes an unused file and modifies some strings and rules to improve the clarity and consistency of the code generation output.

### <samp>🤖 Generated by Copilot at 9535c5e</samp>

> _To generate code for components_
> _We updated some files and contents_
> _We changed `nodejs` version_
> _And added some headers for session_
> _We also improved some checks and comments_

### <samp>🤖 Generated by Copilot at 9535c5e</samp>

*  Update nodejs version to 18.16.0 in `.tool-versions` file ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-751af1a340658c7b8176fe32d7db9fadbe15d1d075daba1919a91df04155bc70L1-R4))
*  Add component_type argument to generate_code function and use it to select checks and templates for different component types ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-57c6ea428b4134afc63a249ce4461df96eb6f49a0e7083c5badc9be262291a87L35-R36), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-1072f68a9f44e052a79715bf3ead28286fa62e091d4df16711de9a55ba87ecc2L12-R38), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-bb92b1907162a142ece31a9550fb28dd055e466a55e7b034c7c464756743ff4fR17-R23), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-7017b494ad167ef6f8c0143e27f3df3e38bd4fbff1d4509db5b67fbdeab6c7cbR14-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-3a4bc073aa4683d67597a5caea715013986100661510a7989d40398a4b4547c7R17-R23), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-127d6be0f356c7d8df8b7493a0677e980c2fe64fb89becdda06118090180f2ecR20-R26))
*  Import templates module in `generate_component_code.py` file and initialize result variable as an empty list ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-1072f68a9f44e052a79715bf3ead28286fa62e091d4df16711de9a55ba87ecc2R5-R6), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-1072f68a9f44e052a79715bf3ead28286fa62e091d4df16711de9a55ba87ecc2L12-R38))
*  Modify auth_example, new_prompt, and component_metadata strings to improve visibility, consistency, and clarity in `generate_component_code.py` file ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-1072f68a9f44e052a79715bf3ead28286fa62e091d4df16711de9a55ba87ecc2L21-R44), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-1072f68a9f44e052a79715bf3ead28286fa62e091d4df16711de9a55ba87ecc2L53-R77), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-eccc11522f1d3578a5754be08082ce0f371f77bc6004a1c6444f3e449aab315fL29-R49))
*  Wrap long lines to follow PEP 8 style guide in `generate_component_code.py`, `generate_for_github_issue.py`, `generate.py`, and `main.py` files ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-1072f68a9f44e052a79715bf3ead28286fa62e091d4df16711de9a55ba87ecc2L47-R71), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-1072f68a9f44e052a79715bf3ead28286fa62e091d4df16711de9a55ba87ecc2L74-R99), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-69bedac25cab56c4653721b18b35683ac9f1958fcf91f090470a07db747ee17bL24-R29), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-57c6ea428b4134afc63a249ce4461df96eb6f49a0e7083c5badc9be262291a87L78-R80), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-57c6ea428b4134afc63a249ce4461df96eb6f49a0e7083c5badc9be262291a87L98-R100), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-57c6ea428b4134afc63a249ce4461df96eb6f49a0e7083c5badc9be262291a87L117-R120), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-38d985e2c8bd73ed773e44c2b976a924b869118a50bef45eabdbe864dff02064L20-R21))
*  Import logging_config module and use logger instead of print statements in `generate_for_github_issue.py` file ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-69bedac25cab56c4653721b18b35683ac9f1958fcf91f090470a07db747ee17bL5-R14), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-69bedac25cab56c4653721b18b35683ac9f1958fcf91f090470a07db747ee17bL41-R53))
*  Add tries argument to generate and main functions and pass it to the model calls in `generate.py` and `main.py` files ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-57c6ea428b4134afc63a249ce4461df96eb6f49a0e7083c5badc9be262291a87L35-R36), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-38d985e2c8bd73ed773e44c2b976a924b869118a50bef45eabdbe864dff02064L28-R30), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-38d985e2c8bd73ed773e44c2b976a924b869118a50bef45eabdbe864dff02064L38-R41))
*  Add comment to explain time.sleep in parse_urls function in `generate.py` file ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-57c6ea428b4134afc63a249ce4461df96eb6f49a0e7083c5badc9be262291a87L98-R100))
*  Change logging level from "WARN" to "DEBUG" in `config.py` file ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-75d5a3dabdf1cce5000f903be0a7a4b5075a7c679f01e23adf164c8486e7167eL37-R37))
*  Change run method signature to only accept { $ } instead of {steps, $ } in action components to follow new convention ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-a02cc62d4388df7fdae8382f11d78c611fbbb97ec98e0edc2261f2b44aafe989L11-R15), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-d0a0fe82bd451e571a186f59b2be01109c9c1d11ffe8cecbb5ff015104d1cbffL25-R28), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-d0a0fe82bd451e571a186f59b2be01109c9c1d11ffe8cecbb5ff015104d1cbffL62-R65))
*  Change version value to use dynamic version of "0.0.{{ts}}" instead of hardcoded timestamp in all components ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-2d4d16cbeaa810ebb19d7ac618bf7d2913dc83ddb3e58833e8e8003061faa7c7L1-R18), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-d0a0fe82bd451e571a186f59b2be01109c9c1d11ffe8cecbb5ff015104d1cbffL1-R16), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-eccc11522f1d3578a5754be08082ce0f371f77bc6004a1c6444f3e449aab315fL7-R9), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-eccc11522f1d3578a5754be08082ce0f371f77bc6004a1c6444f3e449aab315fL19-R23))
*  Add headers to auth_example, export_summary, main_example, other_example, app_prop, rules, and introduction strings to improve visibility and consistency in code generation output ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-1072f68a9f44e052a79715bf3ead28286fa62e091d4df16711de9a55ba87ecc2L21-R44), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-a00c7a38399205f06acde15d8f6059ba01a36f746957b98756ed5b926011df1eL1-R3), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-2d4d16cbeaa810ebb19d7ac618bf7d2913dc83ddb3e58833e8e8003061faa7c7L1-R18), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-d0a0fe82bd451e571a186f59b2be01109c9c1d11ffe8cecbb5ff015104d1cbffL1-R16), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-4586880b75834684fa792214bbd83145b38711fc47db6373a06603e22708c1e1L1-R4), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-7bf583913103650b0a025811d63a3807f1580af6ea2a78fffbdd935f0a232d45L3-R85), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-da06ff0b4bf9a16eea3ca946f01d021ef42c6d71246fb3d94add26cb64ee3b99L3-R3), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-ca2841e93701f19f82d53814ca69a2728eede94ebd31a57d7b2c02d5fd9c7bd1L1-R9))
*  Add example code and explanation to rule 2 and add new rules 9 and 9b to specify naming conventions in rules string ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-da06ff0b4bf9a16eea3ca946f01d021ef42c6d71246fb3d94add26cb64ee3b99L9-R10), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-da06ff0b4bf9a16eea3ca946f01d021ef42c6d71246fb3d94add26cb64ee3b99L29-R27), [link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-da06ff0b4bf9a16eea3ca946f01d021ef42c6d71246fb3d94add26cb64ee3b99L63-R71))
*  Add other_example template to provide an alternative example of an action component that makes a test request against the Slack API ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-bb92b1907162a142ece31a9550fb28dd055e466a55e7b034c7c464756743ff4fR29-R30))
*  Move other_example template to the templates/actions folder to organize the templates by component type ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-bb92b1907162a142ece31a9550fb28dd055e466a55e7b034c7c464756743ff4fL34-L35))
*  Delete additional_rules.py file from the templates/sources/polling folder as it was empty and not used ([link](https://github.com/PipedreamHQ/pipedream/pull/8325/files?diff=unified&w=0#diff-0b9bf62c3d8c42309bac529b815bea365722a93ed4644130244f6a4b3b3e7626L7))

### CODE REVIEW

File: packages/component_code_gen/code_gen/generate.py [hunk 1]
```diff
@@ -65,56 +70,46 @@ def parse_common_files(app, component_type, custom_path=None):
...
 
 
-def parse_urls(urls):
-    contents = []
-    if len(urls) == 0:
-        return contents
-
-    if not config["browserless"]["api_key"]:
-        raise Exception("Missing required browserless api key in config")
+def init_driver(api_key):
```
Comment (dylburger) [inline packages/component_code_gen/code_gen/generate.py:73]: Separated the initialization of the browser from the parsing of URLs

File: packages/component_code_gen/helpers/langchain_helpers.py [hunk 2]
```diff
@@ -80,8 +80,8 @@ def create_user_prompt(prompt, urls_content):
         return prompt
 
     user_prompt = f"{prompt}\n\n## API docs\n\n"
-    for url, content in urls_content:
-        user_prompt += f"\n\n### {url}\n\n{content}"
+    for item in urls_content:
+        user_prompt += f"\n\n### {item['url']}\n\n{item['content']}"
```
Comment (dylburger) [inline packages/component_code_gen/helpers/langchain_helpers.py:84]: `for url, content in urls_content` was actually failing to pass docs content before — that will not iterate over the key and value of dictionary items.

File: packages/component_code_gen/config/config.py [hunk 3]
```diff
@@ -34,6 +34,6 @@ def get_env_var(var_name, required=False, default=None):
         "api_key": get_env_var("SUPABASE_API_KEY", required=True),
     },
     "logging": {
-        "level": get_env_var("LOGGING_LEVEL", default="WARN"),
+        "level": get_env_var("LOGGING_LEVEL", default="DEBUG"),
```
Comment (andrewjschuang) [inline packages/component_code_gen/config/config.py:37]: Default will be `verbose`?

File: packages/component_code_gen/helpers/langchain_helpers.py (hunk 2 above)
Comment (andrewjschuang) [reply packages/component_code_gen/helpers/langchain_helpers.py:84]: Nice! That’s probably the reason the output got worse!

**Review by andrewjschuang — APPROVED** [review]
LGTM!

**Pull request merged**